client/hotrod-client/src/main/java/org/infinispan/client/hotrod/near/NearCacheService.java

```diff
@@ -49,17 +49,22 @@ public void start(RemoteCache<K, V> remote) {
          // Create near cache
          cache = createNearCache(config);
          // Add a listener that updates the near cache
-         listener = createListener(remote);
-         remote.addClientListener(listener);
+         createAndRegisterListener(remote);
          // Get the listener ID for faster listener connected lookups
          listenerId = listenerNotifier.findListenerId(listener);
       }
    }
 
-   private Object createListener(RemoteCache<K, V> remote) {
-      return config.mode().invalidated()
-            ? new InvalidatedNearCacheListener(this)
-            : new EagerNearCacheListener(this, remote.getRemoteCacheManager().getMarshaller());
+   private void createAndRegisterListener(RemoteCache<K, V> remote) {
+      // Attempt to register the smart invalidation listener
+      listener = new KeyOnlyInvalidatedNearCacheListener<>(this, remote.getRemoteCacheManager().getMarshaller());
```

Comment:
I am not a big fan of this, the `ClientCacheEntryCreatedEvent`, `ClientCacheEntryModifiedEvent`, `ClientCacheEntryRemovedEvent` & `ClientCacheEntryExpiredEvent` just allow access to the key and possibly version. Thus these should always not contain the value for an event. This would then make all listeners faster as well, not just near cache.

Assuming this is in place then we just need to remove the `handleCreatedEvent` method that was there before.

I will look at this a little bit to see if there is an easy way to do this.

Reply:
I was fooled by https://github.com/infinispan/infinispan/blob/1560df5ee5e36d7ac4eee784fd1663af39768d00/server/hotrod/src/main/java/org/infinispan/server/hotrod/ClientListenerRegistry.java#L383 passing the value, which is then ignored.
However for the server-side event, we do want to avoid sending values around.

Reply:
Actually, wouldn't a flag on the listener, akin to IGNORE_RETURN_VALUES, be useful in the more general use-case ?

Reply:
I added

```   
public void testNoConverterFactoryCustomEvents() {
      NoConverterFactoryListener l = new NoConverterFactoryListener<>(remoteCacheManager.getCache());
      withClientListener(l, remote -> {
         l.expectNoEvents();
         remote.put(1, "one");
         l.expectCreatedEvent(new CustomEvent(1, "one", 0));
      });
   }
```
to the class `ClientCustomEventsTest.java` and when I debug I see that it creates a `CreatedEventImpl` which is the non custom form. This actually causes no event to be seen at all by the client then.

Reply:
I should have some changes for this pretty soon.

Reply:
Having some issues with the whole key and value data conversions now though - have to keep plugging.